GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -423,6 +441,25 @@ private void SetRowCount(int count)
                 {
                     RowCount = count;
                 }
+
+                if (ToBeSelectedRowIndexes.Any() &&
+                    ToBeSelectedRowIndexes.TryPeek(out int toBeSelectedRowIndex) &&
+                    count > toBeSelectedRowIndex)
```

Comment:
Calling `Any` here is redundant, and likely allocates a few objects. `TryPeek` returns `false` if the collection is empty.

Reply:
Fixed directly in master